src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -688,7 +688,10 @@ ApiGlobe.prototype.pan = function pan(pVector) {
  * @return     {number}  The zoom level.
  */
 ApiGlobe.prototype.getZoomLevel = function getZoomLevel() {
-    return this.scene.getMap().getZoomLevel();
+    const theoricalZoom = this.scene.getMap().computeTheoricalZoom(this.scene.currentCamera(), this.getRange());
+    // const idNode = this.scene.gfxEngine.screenCoordsToNodeId();
+    // const realZoom = this.scene.getMap().getZoomLevelFromIdnode(idNode);
+    return theoricalZoom;
```

Comment:
>when I do a setZoomLevel(X) and then a getZoomLevel(), it does not always return X, but sometimes X+1 (ex : setZoomLevel(1) -> getZoomLevel() = 2

In 3D, there is no concept of getZoomLevel and setZoomLevel as in 2D, because we can have many LoDs in scene depending on tile position. If we must implement set and getZoomLevel, it may  have the problem as you said.